dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/CDIContainerImpl.java

```diff
@@ -632,10 +632,28 @@ private ExtensionArchive newSPIExtensionArchive(ServiceReference<CDIExtensionMet
         Bundle bundle = sr.getBundle();
 
         Set<Class<? extends Extension>> extensionClasses = webSphereCDIExtensionMetaData.getExtensions();
+        Set<Class<?>> beanClasses = webSphereCDIExtensionMetaData.getBeanClasses();
+
+        for (Iterator<Class<? extends Extension>> i = extensionClasses.iterator(); i.hasNext();) {
+            Class extensionClass = i.next();
+            if (extensionClass.getClassLoader() != webSphereCDIExtensionMetaData.getClass().getClassLoader()) {
+                i.remove();
+                Tr.error(tc, "spi.extension.class.in.different.bundle.CWOWB1011E", extensionClass.getCanonicalName());
+            }
+        }
```

Comment:
If we're adding this requirement, it should be added to the javadoc for `CDIExtensionMetadata`.

Currently it says that the result of `getBeanClasses()` must be in the same archive, but doesn't say this for `getExtensionClasses()`.